src/presentation/http/router/join.ts

```diff
@@ -0,0 +1,74 @@
+import type { FastifyPluginCallback } from 'fastify';
+import type NoteSettingsService from '@domain/service/noteSettings.js';
+import type { TeamMember } from '@domain/entities/team.js';
+
+/**
+ * Represents AI router options
+ */
+interface JoinRouterOptions {
+  /**
+   * Note settings service instance
+   */
+  noteSettings: NoteSettingsService
+}
+
+/**
+ * Join Router plugin
+ *
+ * @todo use different replies for different errors in post route
+ * @param fastify - fastify instance
+ * @param opts - router options
+ * @param done - done callback
+ */
+const JoinRouter: FastifyPluginCallback<JoinRouterOptions> = (fastify, opts, done) => {
+  const noteSettingsService = opts.noteSettings;
+
+  fastify.post<{
+    Params: {
+      hash: string
+    }
+  }>('/:hash', {
+    schema: {
+      params: {
+        hash: {
+          $ref: 'JoinSchemaParams#/properties/hash',
+        },
+      },
+      response: {
+        '2xx': {
+          description: 'Team member',
+          content: {
+            'application/json': {
+              schema: {
+                $ref: 'JoinSchemaResponse#/properties',
+              },
+            },
+          },
+        },
+      },
+    },
+    config: {
+      policy: [
+        'authRequired',
+      ],
```

Comment:
there should be also a check for write permission. You can leave a todo for implementing that policy

Reply:
added todo